### ISSUE TITLE
Flat Table [Accessibility] "Form element does not have an implicit (wrapped) <label> & explicit <label>

### ISSUE THREAD

**Issue by divyajindel** [Sage/carbon#4490]
### Current behaviour

Found Accessibility Violation under flat table.

Steps to Reproduce:

Go to https://carbon.sage.com/?path=/story/welcome--welcome-page
After that go to Flat Table ---> with custom column width
and Click on accessibility tab, you can see the violation

[
<img width="902" alt="Screenshot 2021-10-14 at 16 17 06" src="https://user-images.githubusercontent.com/88449494/137347140-1a2400c0-0e3a-4846-a512-0662d974fba0.png">
](url)

### Expected behaviour

We should ensures that every form element has a correct label.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/flat-table--with-custom-column-width

### JIRA Ticket (Sage Only)

FE-4405

### Carbon Version

93.0.2

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.
